ibc-testkit/src/hosts/tendermint.rs

```diff
@@ -0,0 +1,218 @@
+use core::str::FromStr;
+
+use ibc::clients::tendermint::client_state::ClientState;
+use ibc::clients::tendermint::consensus_state::ConsensusState;
+use ibc::clients::tendermint::types::proto::v1::Header as RawHeader;
+use ibc::clients::tendermint::types::{Header, TENDERMINT_HEADER_TYPE_URL};
+use ibc::core::client::types::Height;
+use ibc::core::host::types::identifiers::ChainId;
+use ibc::core::primitives::prelude::*;
+use ibc::core::primitives::Timestamp;
+use ibc::primitives::proto::Any;
+use ibc::primitives::ToVec;
+use tendermint::block::Header as TmHeader;
+use tendermint::validator::Set as ValidatorSet;
+use tendermint_testgen::light_block::TmLightBlock;
+use tendermint_testgen::{
+    Generator, Header as TestgenHeader, LightBlock as TestgenLightBlock,
+    Validator as TestgenValidator,
+};
+
+use super::{TestBlock, TestHeader, TestHost};
+use crate::fixtures::clients::tendermint::ClientStateConfig;
+use crate::testapp::ibc::core::types::MockClientConfig;
+
+#[derive(Debug)]
+pub struct Host(ChainId);
+
+impl TestHost for Host {
+    type Block = TendermintBlock;
+    type BlockParams = BlockParams;
+    type LightClientParams = MockClientConfig;
+    type ClientState = ClientState;
+
+    fn with_chain_id(chain_id: ChainId) -> Self {
+        Self(chain_id)
+    }
+
+    fn chain_id(&self) -> &ChainId {
+        &self.0
+    }
+
+    fn generate_block(
+        &self,
+        height: u64,
+        timestamp: Timestamp,
+        params: &Self::BlockParams,
```

Comment:
Suggestion: What if we introduce getter/setter methods for the "params" instead of passing it as an argument? This would entail adding the relevant fields to the `Host` struct as well. Likewise for the `LightClientParams`.

Reply:
Totally see your point. This should be part of `Host` (or `MockContext`).